src/gennylib/src/quiesce.cpp

```diff
@@ -94,51 +93,93 @@ bool waitOplog(v1::Topology& topology) {
+
+        void onBeforeReplSet(const v1::ReplSetDescription& desc) override {
+            mongocxx::client client(mongocxx::uri(desc.primaryUri));
+            _successAcc = checkCollectionsTestDB(client);
+        }
+
+        void onStandaloneMongod(const v1::MongodDescription& desc) override {
+            mongocxx::client client(mongocxx::uri(desc.mongodUri));
+            _successAcc = checkCollectionsTestDB(client);
+        }
+
+        bool success() {
+            return _successAcc;
+        }
+
+    private:
+        bool checkCollections(mongocxx::database db) {
+            using bsoncxx::builder::basic::kvp;
+            using bsoncxx::builder::basic::make_document;
+
+            auto res = db.run_command(make_document(kvp("serverStatus", 1)));
+            auto idents = res.view()["storageEngine"]["dropPendingIdents"];
+
+            return idents.get_int64() != 0;
+        }
+
+        bool checkCollectionsTestDB(mongocxx::client& client) {
+            auto db = client.database(_dbName);
+            int retries = 0;
+
+            while (checkCollections(db) && retries < DROPPED_COLLECTION_RETRIES) {
+                BOOST_LOG_TRIVIAL(debug)
+                    << "Sleeping 1 second while waiting for collection to finish dropping.";
+                retries++;
+                _sleepContext.sleep_for(std::chrono::seconds(1));
+            }
+            if (retries >= DROPPED_COLLECTION_RETRIES) {
+                BOOST_LOG_TRIVIAL(error) << "Timeout on waiting for collections to drop. "
+                                         << "Tried " << retries << " >= " << DROPPED_COLLECTION_RETRIES
+                                         << " max.";
+                return false;
+            }
+            return true;
+        }
+
+        bool _successAcc = true;
+        std::string _dbName;
+        SleepContext _sleepContext;
+    };
+
+    CheckForDroppedCollectionsVisitor CheckForDroppedCollectionsVisitor(dbName, sleepContext);
+    topology.accept(CheckForDroppedCollectionsVisitor);
+
+    return CheckForDroppedCollectionsVisitor.success();
+}
+
 void doFsync(v1::Topology& topology) {
     class DoFsyncVisitor : public v1::TopologyVisitor {
-        void onMongod(const v1::MongodDescription& desc) override {
+        void onMongod(const v1::MongodDescription& desc) {
             using bsoncxx::builder::basic::kvp;
             using bsoncxx::builder::basic::make_document;
 
             mongocxx::client client(mongocxx::uri(desc.mongodUri));
             auto admin = client["admin"];
             admin.run_command(make_document(kvp("fsync", 1)));
         }
+
+        void onStandaloneMongod(const v1::MongodDescription& desc) override {
+            onMongod(desc);
+        };
+        void onReplSetMongod(const v1::MongodDescription& desc) override {
+            onMongod(desc);
+        };
+        void onConfigSvrMongod(const v1::MongodDescription& desc) override {
+            onMongod(desc);
```

Comment:
(nit) Instead of calling `onMongod` so many times, consider a visitor method `onMongod` in topology.hpp that is called unconditionally for a mongod regardless of type.

Reply:
Good idea, will do